utils/tokenutils.go

```diff
@@ -0,0 +1,89 @@
+package utils
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+type TokenUtil struct {
+	jwtKey []byte
+	//Blocklist key is the JWT and the value is the expiration epoch time
+	blocklist *sync.Map
+}
+
+var ErrExpiredToken = errors.New("Token is expired")
+
+func NewTokenUtil() (*TokenUtil, error) {
+	jwtKey, err := GenerateRandomToken(256)
+	if err != nil {
+		return &TokenUtil{}, err
+	}
+	return &TokenUtil{jwtKey, new(sync.Map)}, nil
+}
+
+func (tu *TokenUtil) CreateJWT(validPeriod time.Duration) (string, error) {
+	claims := jwt.StandardClaims{
+		// In JWT, the expiry time is expressed as unix time
+		ExpiresAt: time.Now().UTC().Add(time.Second * validPeriod).Unix(),
```

Comment:
`validPeriod` should be a complete `time.Duration`.
Update all the callers to use the correct value.
```suggestion
		ExpiresAt: time.Now().UTC().Add(validPeriod).Unix(),
```

Reply:
done -> this is a good idea for user customizability